sqlite/src/lib.rs

```diff
@@ -126,21 +160,46 @@ impl ConnectionWrapper {
             .await
             .unwrap()?;
         Ok(ConnectionWrapper {
-            conn: Arc::new(Mutex::new(conn)),
+            conn: Arc::new(Mutex::new(Some(conn))),
         })
     }
     /// Interact with the underlying SQLite connection. This function
     /// expects a closure that takes the connection as parameter. The
     /// closure is executed in a separate thread so that the async runtime
     /// is not blocked.
-    pub async fn interact<F, R>(&self, f: F) -> R
+    pub async fn interact<F, R>(&self, f: F) -> Result<R, InteractError>
     where
-        F: FnOnce(&rusqlite::Connection) -> R + Send + 'static,
+        F: FnOnce(&rusqlite::Connection) -> Result<R, rusqlite::Error> + Send + 'static,
         R: Send + 'static,
     {
-        let conn = self.conn.clone();
-        tokio::task::spawn_blocking(move || f(&*conn.lock().unwrap()))
-            .await
-            .unwrap()
+        let arc = self.conn.clone();
+        tokio::task::spawn_blocking(move || {
+            let guard = arc.lock().unwrap();
+            let conn = guard.as_ref().unwrap();
+            f(&conn)
+        })
+        .await
+        .map_err(|e| {
+            if e.is_panic() {
+                InteractError::Panic(e.into_panic())
+            } else if e.is_cancelled() {
+                InteractError::Aborted
+            } else {
+                unreachable!();
+            }
```

Comment:
This seems unnecessarily error-prone. Why do this when you can write the panic-safe:

```rust

            if e.is_panic() {
                InteractError::Panic(e.into_panic())
            } else {
                InteractError::Aborted
            }
```

Reply:
I added that last `else` block if there is ever the chance that a third option is added.

But then... this is unlikely ... yet it would generate a meaningful panic. :thinking:

Reply:
Sure, but you would be making your dependent's code panic without 1) documentation and 2) it being a necessity. If you really want to find out if this happens, you _could_ insert a `debug_assert!` instead with a useful message to the user if it is reached. Then at least the panic won't happen in production. But even then, I see little to no value as the no-panic alternative is logically correct.